src/main/java/slimeknights/tconstruct/tools/logic/ToolEvents.java

```diff
@@ -189,25 +186,27 @@ static void onHarvest(ToolHarvestEvent event) {
     }
   }
 
-  static float enderDragonDamage(DamageSource source, LivingEntity entity, float amount) {
+  static void enderDragonDamage(LivingHurtEvent event) {
+    DamageSource source = event.getSource();
+    LivingEntity entity = event.getEntity();
+    float amount = event.getAmount();
     if (!Config.COMMON.dropDragonScales.get()) {
-      return amount;
+      return;
     }
     // dragon being damaged
-//    LivingEntity entity = event.getEntityLiving();
     if (entity.getType() == EntityType.ENDER_DRAGON && amount > 0 && !entity.level().isClientSide) {
       // player caused explosion, end crystals and TNT are examples
-//      DamageSource source = event.getSource();
       if (source.is(DamageTypes.PLAYER_EXPLOSION) && source.getEntity() != null && source.getEntity().getType() == EntityType.PLAYER) {
         // drops 1 - 8 scales
         ModifierUtil.dropItem(entity, new ItemStack(TinkerModifiers.dragonScale, 1 + entity.level().random.nextInt(8)));
       }
     }
-    return amount;
   }
 
-  static float livingAttack(DamageSource source, LivingEntity entity, float amount) {
-//    LivingEntity entity = event.getEntityLiving();
```

Comment:
I'd rather have this code match the forge version linked [here](https://github.com/SlimeKnights/TinkersConstruct/blob/8c347ec456a755374e142db16a7d95a5964b6743/src/main/java/slimeknights/tconstruct/tools/logic/ToolEvents.java#L177) to avoid merge conflicts and make the code for or less the same